### ISSUE TITLE
We should not create and publish sync committee if node is syncing

### ISSUE THREAD

**Issue by tbenr** [Consensys/teku#8005]
I stopped EL, node switched to syncing mode. We correctly stopped producing attestations but we continued contributing to sync committee.
Is it also related to #7862 ?
```
{"@timestamp":"2024-02-20T14:45:40,001","level":"WARN","thread":"validator-async-6","class":"teku-validator-log","message":"Validator   *** Skipped producing attestation while node is syncing  Count: 320, Slot: 7661028","throwable":""}
{"@timestamp":"2024-02-20T14:45:40,007","level":"INFO","thread":"validator-async-0","class":"teku-validator-log","message":"Validator   *** Published sync_signature     Count: 7, Slot: 7661028, Root: 630c707447131405438f1e418ed82dbfd6b5a933e88c394282e27b3dbe87c198","throwable":""}
{"@timestamp":"2024-02-20T14:45:41,870","level":"ERROR","thread":"pool-4-thread-346","class":"TimeBasedEth1HeadTracker","message":"Failed to update eth1 chain head - Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:43,870","level":"ERROR","thread":"pool-4-thread-346","class":"TimeBasedEth1HeadTracker","message":"Failed to update eth1 chain head - Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:44,000","level":"WARN","thread":"ValidatorTimingChannel-0","class":"teku-validator-log","message":"Validator   *** Skipped producing aggregate while node is syncing  Count: 15, Slot: 7661028","throwable":""}
{"@timestamp":"2024-02-20T14:45:45,871","level":"ERROR","thread":"pool-4-thread-346","class":"TimeBasedEth1HeadTracker","message":"Failed to update eth1 chain head - Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:47,872","level":"ERROR","thread":"pool-4-thread-346","class":"TimeBasedEth1HeadTracker","message":"Failed to update eth1 chain head - Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:48,023","level":"INFO","thread":"TimeTick-1","class":"teku-event-log","message":"Syncing     *** Target slot: 7661029, Head slot: 7660932, Remaining slots: 97, Connected peers: 64","throwable":""}
{"@timestamp":"2024-02-20T14:45:48,024","level":"ERROR","thread":"beaconchain-async-35","class":"ForkChoice","message":"Failed to update fork choice. tech.pegasys.teku.ethereum.executionclient.response.InvalidRemoteResponseException: Invalid remote response from the execution client: Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:49,206","level":"WARN","thread":"sync-async-3","class":"BatchSync","message":"Unable to import blocks because execution client is offline.","throwable":""}
{"@timestamp":"2024-02-20T14:45:49,873","level":"ERROR","thread":"pool-4-thread-346","class":"TimeBasedEth1HeadTracker","message":"Failed to update eth1 chain head - Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:51,873","level":"ERROR","thread":"pool-4-thread-346","class":"TimeBasedEth1HeadTracker","message":"Failed to update eth1 chain head - Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:52,000","level":"WARN","thread":"validator-async-2","class":"teku-validator-log","message":"Validator   *** Skipped producing attestation while node is syncing  Count: 327, Slot: 7661029","throwable":""}
{"@timestamp":"2024-02-20T14:45:52,004","level":"INFO","thread":"validator-async-3","class":"teku-validator-log","message":"Validator   *** Published sync_signature     Count: 7, Slot: 7661029, Root: 630c707447131405438f1e418ed82dbfd6b5a933e88c394282e27b3dbe87c198","throwable":""}
{"@timestamp":"2024-02-20T14:45:53,874","level":"ERROR","thread":"pool-4-thread-346","class":"TimeBasedEth1HeadTracker","message":"Failed to update eth1 chain head - Failed to connect to /127.0.0.1:8551","throwable":""}
{"@timestamp":"2024-02-20T14:45:54,301","level":"WARN","thread":"sync-async-3","class":"BatchSync","message":"Unable to import blocks because execution client is offline.","throwable":""}
{"@timestamp":"2024-02-20T14:45:55,523","level":"WARN","thread":"powchain-async-1617","class":"teku-status-log","message":"Eth1 service down or still syncing for 120s, retrying","throwable":""}
```

**Comment by courtneyeh** [comment]
Let me know if there is anymore information that would help me replicate this bug.

I have tried the following:
- Stopping Teku for 10 minutes so it will need to sync when starting back up. Once stated back up I stopped the EL, and while it was syncing, it was not participating or producing anything.
- Stopping the EL so that it would eventually be forced into syncing, but it was still publishing aggregates and sync contributions, and had not started syncing.
- Various combinations surrounding this (of turning off and on the EL and Teku) to try recreate.

**Comment by rolfyone** [comment]
@tbenr follow up?

**Comment by tbenr** [comment]
I successfully replicated in local interop with 64 validators and 2 nodes running 32 validators each, stubbed EL.

one node runs this https://github.com/tbenr/teku/commit/bf9c10642489b8d40340f7c25ff9c240e00c3e9d

1. I let them run stable for a while, in sync
2. I put a breakpoint on the modified node and set `el_online=false`
3. wait the node to fall into sync mode (it takes a while)

here are the logs when that node switches in sync: it stops generating attestations and aggregations but it continues with sync_contribution and sync_signature

```
2024-04-11 14:56:14.007 INFO  - Slot Event  *** Slot: 140, Block:                                                        ... empty, Justified: 12, Finalized: 11, Peers: 1
2024-04-11 14:56:14.010 INFO  - Calling local execution layer to start block production (block slot: 141)
2024-04-11 14:56:14.011 ERROR - Failed to update fork choice. 
java.util.concurrent.CompletionException: java.lang.RuntimeException: stub is offline
...
2024-04-11 14:56:15.503 INFO  - Validator   *** Published aggregate          Count: 1, Slot: 140, Root: e95859ce50e25a735d003f531b559b5bb32203e86fe119da6ccd3cb4c52febe2
2024-04-11 14:56:15.565 INFO  - Validator   *** Published sync_contribution  Count: 32, Slot: 140, Root: e95859ce50e25a735d003f531b559b5bb32203e86fe119da6ccd3cb4c52febe2
2024-04-11 14:56:17.010 INFO  - Creating unsigned block for slot 141
2024-04-11 14:56:17.018 WARN  - Slow Block Production *** Slot: 141 start 10ms, preparation_on_tick +0ms, preparation_apply_deferred_attestations +0ms, preparation_process_head +2ms, retrieve_state +1ms, beacon_block_prepared +5ms, complete +0ms total: 8ms
2024-04-11 14:56:17.019 ERROR - Validator   *** Failed to produce block  Slot: 141 Validator: 88c141d
java.lang.RuntimeException: stub is offline
...
2024-04-11 14:56:18.088 INFO  - Syncing started
2024-04-11 14:56:18.088 WARN  - Stopping eth2 gossip while node is syncing
2024-04-11 14:56:18.428 ERROR - Error while validating payload
java.util.concurrent.CompletionException: java.lang.RuntimeException: stub is offline
...
2024-04-11 14:56:18.439 WARN  - Failed to import block from peer (err: FAILED_EXECUTION_PAYLOAD_EXECUTION) SignedBeaconBlockDeneb{message=BeaconBlockDeneb{slot=109, proposer_index=23, 
...
2024-04-11 14:56:18.505 INFO  - Syncing     *** Target slot: 141, Head slot: 107, Remaining slots: 34, Connected peers: 1
2024-04-11 14:56:18.669 WARN  - Validator   *** Skipped producing attestation while node is syncing  Count: 8, Slot: 141
2024-04-11 14:56:18.678 WARN  - Not publishing sync committee message because no gossip subscriptions are active for slot 141
...
2024-04-11 14:56:18.682 INFO  - Validator   *** Published sync_signature     Count: 64, Slot: 141, Root: e95859ce50e25a735d003f531b559b5bb32203e86fe119da6ccd3cb4c52febe2
2024-04-11 14:56:20.343 WARN  - Validator   *** Skipped producing aggregate while node is syncing  Count: 1, Slot: 141
2024-04-11 14:56:20.583 WARN  - Not publishing sync committee contribution because no gossip subscriptions are active for slot 141
2024-04-11 14:56:20.583 WARN  - Not publishing sync committee contribution because no gossip subscriptions are active for slot 141
2024-04-11 14:56:20.583 WARN  - Not publishing sync committee contribution because no gossip subscriptions are active for slot 141
2024-04-11 14:56:20.584 WARN  - Not publishing sync committee contribution because no gossip subscriptions are active for slot 141
2024-04-11 14:56:20.588 INFO  - Validator   *** Published sync_contribution  Count: 36, Slot: 141, Root: e95859ce50e25a735d003f531b559b5bb32203e86fe119da6ccd3cb4c52febe2
2024-04-11 14:56:22.003 INFO  - Syncing     *** Target slot: 142, Head slot: 107, Remaining slots: 35, Connected peers: 1
2024-04-11 14:56:22.004 ERROR - Failed to update fork choice. 
```

**Comment by rolfyone** [comment]
Thanks for the steps! If someone can please take another look at this, Courtney has moved to other things so anyone should feel free to pick this up please.

**Comment by tbenr** [comment]
I spent more time on it and I found 2 different problems:
1. the production of the sync committee contribution, on the BN side, should be avoided is the node is syncing.
2. the production of the sync message is done by VC independently, without a BN interaction.


for 1. the solution is simple.

for 2. the issue is that VC doesn't have any logic (`ChainHeadTracker`) to understand if its current view on what is the head is based on a event stream that is connected to a syncing node. We should subscribe to `sync_state` and track this in `ChainHeadTracker`
